wsutil/compress.go

```diff
@@ -0,0 +1,193 @@
+package wsutil
+
+import (
+	"bytes"
+	"compress/flate"
+	"errors"
+	"io"
+)
+
+var (
+	ErrWriteClose            = errors.New("write to closed writer")
+	ErrUnexpectedEndOfStream = errors.New("websocket: internal error, unexpected bytes at end of flate stream")
+
+	// Tail as described here: https://tools.ietf.org/html/rfc7692#section-7.2.2
+	deflateFinal = [4]byte{
+		// Bytes from RFC
+		0, 0, 0xff, 0xff,
+	}
+	// Tail to prevent reader error
+	tail = []byte{0x01, 0, 0, 0xff, 0xff}
+)
+
+type CompressorReader interface {
+	io.ReadCloser
+
+	Reset(io.Reader, []byte)
+}
+
+type compressReader struct {
+	flateReader io.ReadCloser
+}
+
+func NewCompressReader(r io.Reader) CompressorReader {
+	return &compressReader{
+		flateReader: flate.NewReader(
+			io.MultiReader(
+				r,
+				bytes.NewReader(deflateFinal[:]),
+				bytes.NewReader(tail[:]),
```

Comment:
`[:]` is redundant 'cause `tail` is already a slice 😉 
or maybe it's better to convert `tail` to an array as a `defalateFinal`

Reply:
Yes, the second option sounds good :)